blocks/blockstore/caching_test.go

```diff
@@ -0,0 +1,35 @@
+package blockstore
+
+import "testing"
+
+func TestCachingOptsLessThanZero(t *testing.T) {
+	opts := DefaultCacheOpts()
+	opts.HasARCCacheSize = -1
+
+	if _, err := CachedBlockstore(nil, nil, opts); err == nil {
+		t.Error("wrong ARC setting was not detected")
```

Comment:
Those test cases are not dependent on each other, so using `Error` all test cases will checked and printed.

`Error = Print + Fail`
`Fatal = Print + FailNow`